src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -246,44 +247,66 @@ private void apiConnectionLogic(HttpServletRequest request, HttpServletResponse
-		nameValues.add(new BasicNameValuePair("content-type", "application/json"));
-		for (final NameValuePair h : nameValues)
-		{
-			clientRequest.addHeader(h.getName(), h.getValue());
-		}
-		BufferedReader rd = null;
-		long startTime = System.currentTimeMillis();
-		try {
-			rd = new BufferedReader(new InputStreamReader(client.execute(clientRequest).getEntity().getContent()));
-		} catch (IOException e) {
-			M_log.error("Canvas API call did not complete successfully", e);
+		if(pathInfo.equalsIgnoreCase(MPATHWAYS_PATH_INFO)){
+			try{
+				M_log.info("MPathways call stub");
+				//Sample File for strict use of testing esb calls made from front end application
+				File testFile = new File( this.getClass().getResource("/mpathwaysSample.txt").toURI() );
+				FileReader fr = new FileReader(testFile);  
+				BufferedReader rd = new BufferedReader(fr);;
+				String line = "";
+				StringBuilder sb = new StringBuilder();
+				while ((line = rd.readLine()) != null) {
+					sb.append(line);
+				}
+				out.print(sb.toString());
+				out.flush();
+			}
+			catch(Exception e){
+				e.printStackTrace();
+			}
+
 		}
-		long stopTime = System.currentTimeMillis();
-		long elapsedTime = stopTime - startTime;
-		M_log.info(String.format("CANVAS Api response took %sms",elapsedTime));
-		String line = "";
-		StringBuilder sb = new StringBuilder();
-		while ((line = rd.readLine()) != null) {
-			sb.append(line);
+		else{
+			String sessionId = request.getSession().getId();
+			String loggingApiWithSessionInfo = String.format("Canvas API request with Session Id \"%s\" for URL \"%s\"", sessionId,url);
+			M_log.info(loggingApiWithSessionInfo);
+			HttpUriRequest clientRequest = null;
+			if(request.getMethod().equals(GET)) {
+				clientRequest = new HttpGet(url);
+			}else if (request.getMethod().equals(POST)) {
+				clientRequest = new HttpPost(url);
+			}else if(request.getMethod().equals(PUT)) {
+				clientRequest=new HttpPut(url);
+			}else if(request.getMethod().equals(DELETE)) {
+				clientRequest=new HttpDelete(url);
+			}
+			HttpClient client = new DefaultHttpClient();
+			final ArrayList<NameValuePair> nameValues = new ArrayList<NameValuePair>();
+			nameValues.add(new BasicNameValuePair("Authorization", "Bearer"+ " " +canvasToken));
+			nameValues.add(new BasicNameValuePair("content-type", "application/json"));
+			for (final NameValuePair h : nameValues)
+			{
+				clientRequest.addHeader(h.getName(), h.getValue());
+			}
+			BufferedReader rd = null;
+			long startTime = System.currentTimeMillis();
+			try {
+				rd = new BufferedReader(new InputStreamReader(client.execute(clientRequest).getEntity().getContent()));
+			} catch (IOException e) {
+				M_log.error("Canvas API call did not complete successfully", e);
+			}
+			long stopTime = System.currentTimeMillis();
+			long elapsedTime = stopTime - startTime;
+			M_log.info(String.format("CANVAS Api response took %sms",elapsedTime));
```

Comment:
this is optional, use `Guava`library for logging time. It puts the unit of time automatically like ms or sec based on the amount of time taken. I just find it very handy. 
```
Stopwatch stopwatch = Stopwatch.createStarted();
stopwatch.stop();
M_log.info("The Api call  took \""+stopwatch+"\" );
in logs it looks like 
The Api call took 3.4ms
The Api call took 34 sec
```